### ISSUE TITLE
What is the configuration of your computer, such as GPU model and GPU memory size

### ISSUE THREAD

**Issue by zhongqiu1245** [zylo117/Yet-Another-EfficientDet-Pytorch#430]
Thank you for your amazing job!
What is the configuration of your computer, such as GPU model and GPU memory size?
I'm looking forward to your reply.

**Comment by zylo117** [comment]
2080ti, 11G

**Comment by zhongqiu1245** [comment]
If I want to train D7, how many GPUs(2080ti, 11G) should I prepare?

**Comment by NikZak** [comment]
I believe that you would not be able to fit the D7 model on 2080 Ti. It does not matter how many as D7 would not fit into RAM of a single 2080Ti (and you need to be able to load whole model into RAM of each GPU) even with batchsize of 1, unless you 
1) train head-only 
2) decrease the default resize resolution of input images in the train.py file
I have a 2080 Ti and I stumbled at D5

**Comment by zylo117** [comment]
@NikZak It seems training in FP16 is much harder for effdet based on my previous experiments.
Most models like yolo, rcnn can performs well in fp16 even if they are trained in fp32. But effdet doesn't. You can try running coco_eval in fp16 with fp32 weights and the mAP will be half of it used to be. Probably because there are too many shared parameters, so they have to be more precise.

You can modify the train.py following the coco_eval.py to train in fp16. BTW, fp16 is not supported when using dataparallel.

**Comment by NikZak** [comment]
Just investigating the memory issue further:
Here is a [thread](https://github.com/google/automl/issues/85) from the official repository of EfficientDet. Looks like memory for GPU training is a big problem: 
- D0 w batch size 8 not trainable on 11Gb GPU
- D5 w batch size 1 not trainable on 11Gb GPU
- D6 w batch size 1 not trainable on 24Gb(!) GPU
However training on TPU does not pose memory problem
- D7 w batch size 4 trainable on TPUv3, where each core has 16Gb

Someone proposes that lines like this in [the code](https://github.com/google/automl/blob/75ba619627959b774fea8caec7b6373ac95e98b5/efficientdet/det_model_fn.py#L238-L269) cause the trouble: 
 # Sum per level losses to total loss. 
 cls_loss = tf.add_n(cls_losses) 
 box_loss = tf.add_n(box_losses) 
as TensorFlow has to keep activations from each layer to aggregate a single gradient

For Tensorflow, there is a probable solution [here](https://stackoverflow.com/questions/36194394/how-i-reduce-memory-consumption-in-a-loop-in-tensorflow) (aggregation_method=tf.AggregationMethod.EXPERIMENTAL_ACCUMULATE_N) but it does not seem to be working as people still complaining. 

However a solution must exist as training works on TPU
